src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.WaitableObject.Unix.cs

```diff
@@ -744,9 +744,9 @@ private void SignalMutex(bool isAbandoned)
                 Debug.Assert(IsMutex);
                 Debug.Assert(!IsSignaled);
 
-                _ownershipInfo.RelinquishOwnership(this, isAbandoned);
+                _ownershipInfo!.RelinquishOwnership(this, isAbandoned);
```

Comment:
Rather than adding these `!`s, can you add a `Debug.Assert(_ownershipInfo != null);` above?